identity/src/keypair.rs

```diff
@@ -372,63 +377,49 @@ impl TryInto<rsa::Keypair> for Keypair {
     type Error = OtherVariantError;
 
     fn try_into(self) -> Result<rsa::Keypair, Self::Error> {
-        #[allow(deprecated)]
-        match self {
-            Keypair::Rsa(inner) => Ok(inner),
+        match self.keypair {
+            Inner::Rsa(inner) => Ok(inner),
             #[cfg(feature = "ed25519")]
-            Keypair::Ed25519(_) => Err(OtherVariantError::new(crate::KeyType::Ed25519)),
+            Inner::Ed25519(_) => Err(OtherVariantError::new(KeyType::Ed25519)),
             #[cfg(feature = "secp256k1")]
-            Keypair::Secp256k1(_) => Err(OtherVariantError::new(crate::KeyType::Secp256k1)),
+            Inner::Secp256k1(_) => Err(OtherVariantError::new(KeyType::Secp256k1)),
             #[cfg(feature = "ecdsa")]
-            Keypair::Ecdsa(_) => Err(OtherVariantError::new(crate::KeyType::Ecdsa)),
+            Inner::Ecdsa(_) => Err(OtherVariantError::new(KeyType::Ecdsa)),
         }
     }
 }
 
-/// The public key of a node's identity keypair.
 #[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub enum PublicKey {
+enum PublicKeyType {
```

Comment:
@thomaseizinger To be consistent with `Inner`, how do you want to rename this `enum`? Maybe this one `PublicKeyInner` and `Inner` becomes `KeyPairInner`?

Reply:
@thomaseizinger Done